packages/experiments/src/components/FloatingPicker/BaseFloatingPicker.tsx

```diff
@@ -367,66 +348,30 @@ export class BaseFloatingPicker<T, P extends IBaseFloatingPickerProps<T>> extend
         break;
 
       case KeyCodes.del:
-        if (this.suggestionStore.currentIndex !== -1) {
-          if (this.props.onRemoveSuggestion) {
-            (this.props.onRemoveSuggestion as ((item: T) => void))(
-              this.suggestionStore.currentSuggestion!.item
-            );
-          }
-          this.suggestionStore.removeSuggestion(
-            this.suggestionStore.currentIndex
+        if (this.props.onRemoveSuggestion
+          && this.suggestionsControl.hasSuggestionSelected
+          && this.suggestionsControl.currentSuggestion) {
+          (this.props.onRemoveSuggestion as ((item: T) => void))(
+            this.suggestionsControl.currentSuggestion!.item
           );
+
+          this.suggestionsControl.removeSuggestion();
           this.forceUpdate();
         }
         ev.stopPropagation();
         break;
 
       case KeyCodes.up:
-        if (suggestionElement && suggestionElement.tryHandleKeyDown(keyCode, this.suggestionStore.currentIndex)) {
+        if (this.suggestionsControl && this.suggestionsControl.handleKeyDown(keyCode)) {
           ev.preventDefault();
           ev.stopPropagation();
-        } else {
-          if (suggestionElement && suggestionElement.hasSuggestedAction() &&
-            this.suggestionStore.currentIndex === 0
-          ) {
-            ev.preventDefault();
-            ev.stopPropagation();
-            suggestionElement.focusAboveSuggestions();
-            this.suggestionStore.deselectAllSuggestions();
-            this.forceUpdate();
-          } else {
-            if (this.suggestionStore.previousSuggestion()) {
-              ev.preventDefault();
-              ev.stopPropagation();
-              this.onSuggestionSelect();
-            }
-          }
         }
         break;
 
       case KeyCodes.down:
-        if (suggestionElement && suggestionElement.tryHandleKeyDown(keyCode, this.suggestionStore.currentIndex)) {
+        if (this.suggestionsControl && this.suggestionsControl.handleKeyDown(keyCode)) {
           ev.preventDefault();
```

Comment:
I really like moving all of this into another component.

Reply:
Yes I think the boundary of responsibility is much cleaner this way